scripts/bulk-index.js

```diff
@@ -400,22 +406,40 @@ const buildSqlQuery = (options) => {
   // Querying by existance of a marc field?
   } else if (options.nyplSource && options.type && options.hasMarc) {
     type = options.type
-    sqlFromAndWhere = `${type},
-      json_array_elements(var_fields::json) jV
-      WHERE nypl_source = $1
-      AND jV->>'marcTag' = $2`
+
+    // Build array of SELECT clauses:
+    const selects = [type, 'json_array_elements(var_fields::json) jV']
+    // Build array of WHERE clauses:
+    const wheres = [
+      'nypl_source = $1',
+      "jV->>'marcTag' = $2"
+    ]
+    // Collect user input:
     params = [
       options.nyplSource,
       options.hasMarc
     ]
+
+    // If filtering on existence of specific subfield, add clause:
+    if (options.hasSubfield) {
+      selects.push("json_array_elements(jV->'subfields') jVS")
+      wheres.push("jVS->>'tag' = $3")
+      params.push(options.hasSubfield)
+    }
+
+    sqlFromAndWhere = selects.join(',\n') +
+      '\nWHERE ' + wheres.join('\nAND ')
   } else {
     throw new Error('Insufficient options to buildSqlQuery')
   }
 
-  const query = `SELECT * FROM ${sqlFromAndWhere}` +
+  const innerSelect = `SELECT DISTINCT id, nypl_source FROM ${sqlFromAndWhere}` +
```

Comment:
can you add a comment here describing why this is necessary?

Reply:
Added note about how and why.